compiler/modules/front/codeGen.bal

```diff
@@ -1116,10 +1115,10 @@ function codeGenExpr(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:Ex
             return codeGenCheckingExpr(cx, bb, env, checkingKeyword, operand, pos);
         }
         var { opPos: pos, bitwiseOp: op, left, right } => {
-            var { result: l, block: block1} = check codeGenExprForInt(cx, bb, env, left);
-            var { result: r, block: nextBlock } = check codeGenExprForInt(cx, block1, env, right);
-            // We evaluate the operands here, so we can reuse the function for compound assignment.
-            return codeGenBitwiseBinaryExpr(cx, nextBlock, op, pos, l, r);
+            var { lhs, rhs, nextBlock, ifNilBlock } = check codeGenBinaryNilLift(cx, env, left, right, bb, pos);
+            bir:IntOperand l = check intOperand(cx, lhs, pos);
```

Comment:
Using the opPos here isn't going to give a very good message. Maybe use Range for left/right expression.

Reply:
fixed